src/posix/client.cpp

```diff
@@ -28,6 +28,8 @@
 
 #include "platform/openthread-posix-config.h"
 
+#include "cli/cli_config.h"
```

Comment:
nit: I think `cli` is an app, and posix/client is another app. Is it proper for one app to include headers of another app?

Reply:
I take `cli` as a text based control protocol of OpenThread, which can be embedded in other apps, like otbr-agent.

Reply:
Do you imply that we should move `cli` to core?

Reply:
No, `cli` is already a standalone library which can be optionally linked to other apps.